ci/openshift-ci/peer-pods-azure.sh

```diff
@@ -0,0 +1,213 @@
+#!/bin/bash -e
+# Setup peer-pods using cloud-api-adaptor on azure
+
+###############################
+# Disable security to allow e2e
+###############################
+
+# Disable security
+oc adm policy add-scc-to-group privileged system:authenticated system:serviceaccounts
+oc adm policy add-scc-to-group anyuid system:authenticated system:serviceaccounts
+oc label --overwrite ns default pod-security.kubernetes.io/enforce=privileged pod-security.kubernetes.io/warn=baseline pod-security.kubernetes.io/audit=baseline
+
+
+####################################
+# Get basic credentials from cluster
+####################################
+
+oc -n kube-system get secret azure-credentials -o json > azure_credentials.json
+AZURE_CLIENT_ID="$(jq -r .data.azure_client_id azure_credentials.json|base64 -d)"
+AZURE_CLIENT_SECRET="$(jq -r .data.azure_client_secret azure_credentials.json|base64 -d)"
+AZURE_TENANT_ID="$(jq -r .data.azure_tenant_id azure_credentials.json|base64 -d)"
+AZURE_SUBSCRIPTION_ID="$(jq -r .data.azure_subscription_id azure_credentials.json|base64 -d)"
+rm -f azure_credentials.json
+AZURE_RESOURCE_GROUP=$(oc get infrastructure/cluster -o jsonpath='{.status.platformStatus.azure.resourceGroupName}')
+az login --service-principal -u "${AZURE_CLIENT_ID}" -p "${AZURE_CLIENT_SECRET}" --tenant "${AZURE_TENANT_ID}"
+
+AZURE_VNET_NAME=$(az network vnet list --resource-group "${AZURE_RESOURCE_GROUP}" --query "[].{Name:name}" --output tsv)
+AZURE_SUBNET_NAME=$(az network vnet subnet list --resource-group "${AZURE_RESOURCE_GROUP}" --vnet-name "${AZURE_VNET_NAME}" --query "[].{Id:name} | [? contains(Id, 'worker')]" --output tsv)
+AZURE_SUBNET_ID=$(az network vnet subnet list --resource-group "${AZURE_RESOURCE_GROUP}" --vnet-name "${AZURE_VNET_NAME}" --query "[].{Id:id} | [? contains(Id, 'worker')]" --output tsv)
+AZURE_REGION=$(az group show --resource-group "${AZURE_RESOURCE_GROUP}" --query "{Location:location}" --output tsv)
+
+# Create workload identity
+AZURE_WORKLOAD_IDENTITY_NAME="caa-${CLUSTER_NAME}"
+az identity create   --name "${AZURE_WORKLOAD_IDENTITY_NAME}"   --resource-group "${AZURE_RESOURCE_GROUP}"   --location "${AZURE_REGION}"
```

Comment:
Per curiosity, how is this resource deleted when it is no longer used ? Same question for other azure resources created by this script.

Reply:
Well I briefly looked at ways to pass this information further the chain (in e2e pipeline) but couldn't find anything straight forward. So I decided to rely on Azure to hopefully destroy everything which is created with temporary subscription. Whether it's true I don't really know as I only used temporary subscription which disappeared so I was unable to check the extra resources were destroyed.